File: server/src/com/bg/discworld/world/Area.java
package com.bg.discworld.world;

import java.util.HashMap;

public class Area {

	public long id = 0;
	public String name = "";
	public String displayName = "";

	public HashMap<Integer, Room> rooms = new HashMap<Integer, Room>();

	// ambient effects, weather data, maybe wild animal likelihood
	// generic area-wide mobile spawns maybe?
	// area-wide item?
	// quest info, maybe list of specific rooms for some purpose

	public Area() {

	}

	public Area(long id) {
		this.id = id;
	}

	public Room searchRoom(String name) {
		for(Room r : rooms.values()) {
			if(r.name.equals(name)) {
				return r;
			}
		}
		return null;
	}
	
	public void update(long tick) {
		for (Room r : rooms.values()) {
			r.update(tick);
		}
	}

}

File: editor/mudeditor/Area.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MUDEdit {
    class Area {

        public String name = "";
        public String displayName = "";

        public Dictionary<int, Room> rooms = new Dictionary<int, Room>();

        public Area() {

        }

        public Area(String name) {
            this.name = name;
            
        }

        public String GetJSON() {
            return Util.ObjectToJSON(this);
        }
    }
}
